pkg/agent/controller/egress/egress_controller.go

```diff
@@ -196,6 +205,18 @@ func NewEgressController(
 	if trafficShapingEnabled && !openflow.OVSMetersAreSupported() {
 		klog.Info("EgressTrafficShaping feature gate is enabled, but it is ignored because OVS meters are not supported.")
 	}
+
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartStructuredLogging(0)
+	runtime.Must(scheme.AddToScheme(k8sscheme.Scheme))
+	eventBroadcaster.StartRecordingToSink(&v1.EventSinkImpl{
+		Interface: client.CoreV1().Events(""),
+	})
```

Comment:
I think typically `StartStructuredLogging` and `StartRecordingToSink` are called in `Run` function, not when initializing.
And `Shutdown` should be deferred in `Run`

Reply:
why it would lead to that? I see they are decoupled objects, I see many code initialize them in constructor then run broadcaster in `Run`.